owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt

```diff
@@ -196,7 +227,142 @@ class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.Sor
             FileListOption.SHARED_BY_LINK -> mainFileListViewModel.getSharedByLinkFilesList()
         }
 
-        // TODO Manage FAB button
+        updateFab(newFileListOption)
+    }
+
+    private fun updateFab(newFileListOption: FileListOption) {
+        if (!newFileListOption.isAllFiles() || isPickingAFolder()) {
+            setFabEnabled(false)
+        } else {
+            setFabEnabled(true)
+            registerFabListeners()
+
+            // detect if a mini FAB has ever been clicked
+            val prefs = PreferenceManager.getDefaultSharedPreferences(activity)
+            if (prefs.getLong(KEY_FAB_EVER_CLICKED, 0) > 0) {
+                miniFabClicked = true
+            }
+
+            // add labels to the min FABs when none of them has ever been clicked on
+            if (!miniFabClicked) {
+                setFabLabels()
+            } else {
+                removeFabLabels()
+            }
+        }
+    }
+
+    /**
+     * Sets the 'visibility' state of the FAB contained in the fragment.
+     *
+     *
+     * When 'false' is set, FAB visibility is set to View.GONE programatically,
+     *
+     * @param enabled Desired visibility for the FAB.
+     */
+    private fun setFabEnabled(enabled: Boolean) {
+        if (enabled) {
+            binding.fabMain.visibility = View.VISIBLE
+        } else {
+            binding.fabMain.visibility = View.GONE
+        }
+    }
```

Comment:
I think that we can reduce this method, like this:

```
    private fun setFabEnabled(enabled: Boolean) 
    {
       binding.fabMain.isVisible = enabled
    }
```
What do you think?

Reply:
👍